relay-server/src/services/processor/replay.rs

```diff
@@ -74,10 +73,13 @@ pub fn process(
                     client_addr,
                     user_agent,
                 ) {
-                    Err(outcome) => ItemAction::Drop(outcome),
+                    Err(outcome) => {
+                        if let Outcome::Invalid(reason) = outcome {
+                            return Err(ProcessingError::Invalid(reason));
+                        }
+                    }
```

Comment:
You can now refactor `handle_replay_event_item` and other helper functions to return `Result<Bytes, DiscardReason>`, then this becomes
```suggestion
                    Err(reason) => {
                        return Err(ProcessingError::Invalid(reason));
                    }
```

Or even (haven't tested this)

```rust
let replay_event = handle_replay_event_item(...).map_err(ProcessingError::Invalid)?;
```